specification/under_development/Instruction_Set.md

```diff
@@ -0,0 +1,52 @@
+
+
+All quantum instructions are represented by LLVM external functions.
+Quantum instructions may take qubits, doubles, or integers as parameters,
+and should all have no return; that is, they should be void.
+
+Measurements should take an offset into the `ClassicalStorage` global as a parameter.
+The measurement result should be stored into the corresponding bit in `ClassicalStorage`.
+
+The LLVM functions that implement the quantum instruction set should all have
+names that start with `__quantum__qis__`.
+
+QIR does not specify the contents of the quantum instruction set.
+However, in order to ensure some amount of uniformity, implementations that provide
+any of the following quantum instructions must match the specified definition:
+
+| Operation Name | LLVM Function Declaration  | Description | Matrix |
+|----------------|----------------------------|-------------|--------|
+| CCx, CCNOT, Toffoli | `__quantum__qis__toffoli__body (%Qubit control1, %Qubit control1, %Qubit target)` | Toffoli or doubly-controlled X | ![latex](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cbegin%7Bbmatrix%7D+1+%26+0+%26+0+%26+0+%26+0+%26+0+%26+0+%26+0+%5C%5C+0+%26+1+%26+0+%26+0+%26+0+%26+0+%26+0+%26+0+%5C%5C+0+%26+0+%26+1+%26+0+%26+0+%26+0+%26+0+%26+0+%5C%5C+0+%26+0+%26+0+%26+1+%26+0+%26+0+%26+0+%26+0+%5C%5C+0+%26+0+%26+0+%26+0+%26+1+%26+0+%26+0+%26+0+%5C%5C+0+%26+0+%26+0+%26+0+%26+0+%26+1+%26+0+%26+0+%5C%5C+0+%26+0+%26+0+%26+0+%26+0+%26+0+%26+0+%26+1+%5C%5C+0+%26+0+%26+0+%26+0+%26+0+%26+0+%26+1+%26+0+%5C%5C+%5Cend%7Bbmatrix%7D) |
+| Cx, CNOT | `__quantum__qis__cnot__body (%Qubit control, %Qubit target)` | CNOT or singly-controlled X | ![latex](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cbegin%7Bbmatrix%7D+1+%26+0+%26+0+%26+0+%5C%5C+0+%26+1+%26+0+%26+0+%5C%5C+0+%26+0+%26+0+%26+1+%5C%5C+0+%26+0+%26+1+%26+0+%5C%5C+%5Cend%7Bbmatrix%7D) |
+| Cz | `__quantum__qis__cz__body (%Qubit control, %Qubit target)` | Singly-controlled Z | ![latex](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cbegin%7Bbmatrix%7D+1+%26+0+%26+0+%26+0+%5C%5C+0+%26+1+%26+0+%26+0+%5C%5C+0+%26+0+%26+1+%26+0+%5C%5C+0+%26+0+%26+0+%26+-1+%5C%5C+%5Cend%7Bbmatrix%7D) |
+| H | `__quantum__qis__h__body (%Qubit q)` | Hadamard | ![latex](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cfrac%7B1%7D%7B%5Csqrt%7B2%7D%7D%5Cbegin%7Bbmatrix%7D+1+%26+1+%5C%5C+1+%26+-1+%5C%5C+%5Cend%7Bbmatrix%7D) |
+| Mz or Measure | `__quantum__qis__mz__body (%Qubit q, i32 result_offset)` | Measure a qubit along the the Pauli Z axis |
```

Comment:
This signature for measure does not use `%Result*` and instead opts for an explicit i32 for the identifier/offset. Is that intentional or should `%Result*` be used instead?

Reply:
This table is outdated. It should be %Result*, matching what is shown in the actual Profile doc.